src/overlays/actors/ovl_Bg_Ice_Shelter/z_bg_ice_shelter.h

```diff
@@ -11,9 +11,17 @@ typedef void (*BgIceShelterActionFunc)(struct BgIceShelter*, PlayState*);
 typedef struct BgIceShelter {
     /* 0x0000 */ DynaPolyActor dyna;
     /* 0x0164 */ BgIceShelterActionFunc actionFunc;
-    /* 0x0168 */ ColliderCylinder cylinder1;
-    /* 0x01B4 */ ColliderCylinder cylinder2;
+    /* 0x0168 */ ColliderCylinder cylinder1; // Used to detect blue fire
+    /* 0x01B4 */ ColliderCylinder cylinder2; // Only used by the types without dynapoly collision
     /* 0x0200 */ s16 alpha;
 } BgIceShelter; // size = 0x0204
 
+typedef enum {
+    /* 0 */ RED_ICE_LARGE,    // Large red ice block
+    /* 1 */ RED_ICE_SMALL,    // Small red ice block
+    /* 2 */ RED_ICE_PLATFORM, // Complex structure that can be climbed and walked on. Unused in vanilla OoT, used in MQ to cover the Ice Cavern Map chest
+    /* 3 */ RED_ICE_WALL,     // Vertical ice sheets blocking corridors
+    /* 4 */ RED_ICE_KING_ZORA // Giant red ice block covering King Zora
+} RedIceType;
```

Comment:
Usually we put these enums more near the top of the .h files, since eventually they should serve as documentation of external use of the actor (not just code-wise)

Reply:
Oh ok, I had written that following the order from `BgBreakwall`. I moved it up.